lib/semantics/resolve-names.cpp

```diff
@@ -5771,6 +5782,14 @@ bool ResolveNamesVisitor::Pre(const parser::ProgramUnit &x) {
   FinishSpecificationParts(root);
   ResolveExecutionParts(root);
   OmpAttributeVisitor{context(), *this}.Walk(x);
+  if (!context().AnyFatalError()) {
+    // The data-sharing attribute of the loop iteration variable for a
+    // sequential loop (2.15.1.1) can only be determined when visiting
+    // the corresponding DoConstruct, a second walk is to adjust the
+    // symbols for all the data-refs of that loop iteration variable
+    // prior to the DoConstruct.
+    OmpAttributeVisitor{context(), *this}.Walk(x);
+  }
```

Comment:
Can you extract this into a separate function so that this function shows the high-level order of processing a program unit?